src/scanner-cli.ts

```diff
@@ -52,31 +52,33 @@ export function normalizePlatformName(): 'windows' | 'linux' | 'macosx' {
 /**
  * Where to download the SonarScanner CLI
  */
-function getScannerCliUrl(properties: ScannerProperties, versionStr: string): URL {
+function getScannerCliUrl(
+  properties: ScannerProperties,
+  versionStr: string,
+  archStr?: string,
+): URL {
   // Get location to download scanner-cli from
 
   // Not in default to avoid populating it for non scanner-cli users
   const scannerCliMirror = properties[ScannerProperty.SonarScannerCliMirror] ?? SCANNER_CLI_MIRROR;
-  const version = semver.coerce(versionStr);
-  if (!version) {
-    throw new Error(`Version "${versionStr}" does not have a correct format."`);
-  }
-  const arch = version.compare('6.1.0') >= 0 ? '-x64' : '';
 
-  const scannerCliFileName =
-    'sonar-scanner-cli-' + versionStr + '-' + normalizePlatformName() + arch + '.zip';
+  let archSuffix = archStr ? `-${archStr}` : '';
+  const scannerCliFileName = `sonar-scanner-cli-${versionStr}-${normalizePlatformName()}${archSuffix}.zip`;
   return new URL(scannerCliFileName, scannerCliMirror);
 }
 
 export async function downloadScannerCli(properties: ScannerProperties): Promise<string> {
-  const version = properties[ScannerProperty.SonarScannerCliVersion] ?? SCANNER_CLI_VERSION;
-  if (!/^[\d.]+$/.test(version)) {
-    throw new Error(`Version "${version}" does not have a correct format."`);
+  const versionStr = properties[ScannerProperty.SonarScannerCliVersion] ?? SCANNER_CLI_VERSION;
+  const version = semver.coerce(versionStr);
+  if (!version) {
+    throw new Error(`Version "${versionStr}" does not have a correct format."`);
   }
+  const archStr = version.compare('6.1.0') >= 0 ? 'x64' : undefined;
+  let archSuffix = archStr ? `-${archStr}` : '';
```

Comment:
NP: should be `const` but feel free to ignore

```suggestion
  const archSuffix = archStr ? `-${archStr}` : '';
```